### DISCUSSION TITLE
Bypass api.github.com rate limit for all: suggestions?

### DISCUSSION THREAD

**Discussion by ivan-hc** [ivan-hc/AM#578] [Ideas]
@nazdridoy what if I use random IPs to call github APIs in curl?

We have a variable in APP-MANAGER:
```
HeaderAuthWithGITPAT=" --header \"Authorization: token $(<"$AMPATH/ghapikey.txt")\" "
```
I've done a test by changing it with
```
HeaderAuthWithGITPAT=' --header "X-Forwarded-for: $((RANDOM % 256)).$((RANDOM % 256)).$((RANDOM % 256)).$((RANDOM % 256))\" '
```
and changed this line in install .am
```
sed -i "s# https://api.github.com#$HeaderAuthWithGITPAT https://api.github.com#g" ./$arg
```
to
```
sed -i "s#curl -Ls https://api.github.com#curl -Ls $HeaderAuthWithGITPAT https://api.github.com#g" ./$arg
```

and tried to install some applications.

I've not reached tha API limit yet, but all the installation scripts had this
```
version=$(curl -Ls --header "X-Forwarded-for: $((RANDOM % 256)).$((RANDOM % 256)).$((RANDOM % 256)).$((RANDOM % 256))" https://api.github.com/repos/$REPO/releases | jq '.' | grep -i $APP | grep browser_download_url | grep -i appimage | cut -d '"' -f 4 | head -1)
```

would this be sufficient to surpass API limit calls for all users?

PS: until now I've always used "[torsocks](https://support.torproject.org/it/glossary/torsocks/)" in case I reached API limits, for example, by updating modules, but its still a slow solution.

The solution above was suggested at https://unix.stackexchange.com/questions/607037/how-to-using-bash-to-generate-random-ips-in-curl but I still have doubts on its real efficiency.

Let me know.

**Comment by ivan-hc** [comment]
UPDATE: just tested a script by myself
```
#!/usr/bin/env bash

while true
do
curl -Ls --header "X-Forwarded-for: $((RANDOM % 256)).$((RANDOM % 256)).$((RANDOM % 256)).$((RANDOM % 256))" https://api.github.com/repos/ivan-hc/Chromium-Web-Browser-appimage/releases | jq '.' | grep -i chromium | grep browser_download_url | grep -i appimage | cut -d '"' -f 4 | head -1
sleep 0.1
done
```
and didn't worked.

**Reply by nazdridoy** [reply]
why would it work?

**Comment by nazdridoy** [comment]
i'm not sure what you are trying to achieve! its vary unlikely for anyone to hit api limit if they use api token..

**Reply by ivan-hc** [reply]
I'd like to include a solution like torsocks... or at least to use a random proxy in "curl", not everyone knows what a api token is and after launching multiple processes (for example by including "AM" into a live media or a distro) the result may be that some scripts will appear as "broken". This has happened to some users in the past.

**Reply by nazdridoy** [reply]
no one has to use api unless they hit rate limit which is very unlikely.... those who do, i want to believe they are savvy enough to use api token. its should be easier then fiddling with proxies...
